### PR TITLE
improve search/results page e2e tests

### PR DESCRIPTION
**Pull request by ijsnow**
Improve search result e2e tests.

### CODE REVIEW

File: web/src/e2e/index.e2e.test.tsx [hunk 1]
```diff
@@ -687,32 +694,61 @@ describe('e2e test suite', function(this: any): void {
...
+            await page.waitForSelector('.e2e-search-results-stats')
             await retry(async () => {
                 const label = await page.evaluate(
-                    () => document.querySelector('.search-results__stats')!.textContent || ''
+                    () => document.querySelector('.e2e-search-results-stats')!.textContent || ''
                 )
                 expect(label.includes('results')).toEqual(true)
             })
+
+            const firstFileMatchHref = await page.$$eval('a.file-match__item', ([a]) => (a as any).href)
```
Comment (felixfbecker) [inline web/src/e2e/index.e2e.test.tsx]:
```suggestion
            const firstFileMatchHref = await page.$eval('a.file-match__item', a => (a as HTMLAnchorElement).href)
```

**Comment by ggilmore** [comment]
Do your tests use the beginning `^` and ending `$` anchors, ex: `repo:^github.com/example/repo^$`?

File: web/src/e2e/index.e2e.test.tsx (hunk 1 above)
Comment (felixfbecker) [reply web/src/e2e/index.e2e.test.tsx]: Could you add a `e2e-` class to the element?

File: web/src/e2e/index.e2e.test.tsx [hunk 2]
```diff
@@ -687,32 +694,61 @@ describe('e2e test suite', function(this: any): void {
...
+
+            const operatorsQuery = Object.keys(operators)
+                .map(op => `${op}:${operators[op]}`)
+                .join('+')
+
+            await page.goto(`${baseURL}/search?q=diff+${operatorsQuery}`)
+            await page.waitForSelector('.e2e-search-results-stats')
+            await retry(async () => {
+                const label = await page.evaluate(
+                    () => document.querySelector('.e2e-search-results-stats')!.textContent || ''
```
Comment (felixfbecker) [inline web/src/e2e/index.e2e.test.tsx]: Maybe also check if at least one result is rendered?

File: web/src/e2e/index.e2e.test.tsx [hunk 3]
```diff
@@ -687,32 +694,62 @@ describe('e2e test suite', function(this: any): void {
...
             // navigate to result on click
-            await page.click('.file-match__item')
+            await page.click('.e2e-file-match-item')
+
             await retry(async () => {
-                expect(await page.evaluate(() => window.location.href)).toEqual(
-                    baseURL +
-                        '/github.com/sourcegraph/go-diff@3f415a150aec0685cb81b73cc201e762e075006d/-/blob/diff/testdata/sample_file_extended_empty_rename.diff#L1:1a'
-                )
+                expect(await page.evaluate(() => window.location.href)).toEqual(firstFileMatchHref)
```
Comment (chrismwendt) [inline web/src/e2e/index.e2e.test.tsx]: Would [`page.waitForNavigation([options])
 `](https://github.com/GoogleChrome/puppeteer/blob/master/docs/api.md#pagewaitfornavigationoptions) be helpful here?

File: web/src/e2e/index.e2e.test.tsx [hunk 4]
```diff
@@ -687,32 +694,62 @@ describe('e2e test suite', function(this: any): void {
...
+                repo: '^github.com/sourcegraph/go-diff$',
+                count: '1000',
+                type: 'file',
+                file: '.go',
+                '-file': '.md',
+            }
+
+            const operatorsQuery = Object.keys(operators)
+                .map(op => `${op}:${operators[op]}`)
+                .join('+')
```
Comment (chrismwendt) [inline web/src/e2e/index.e2e.test.tsx]: Could the `....join('+')` be replaced with `encodeURIComponent(....join(' '))`?

Comment (ijsnow) [reply web/src/e2e/index.e2e.test.tsx]: It could, but it's not necessary

**Pull request merged**